database/errors.go

```diff
@@ -14,3 +17,38 @@ func IsErrConstraintUnique(err error) bool {
 	}
 	return false
 }
+
+// IsErrRetryable returns true if the given error might be
+// transient and the interaction can be safely retried.
+func IsErrRetryable(err error) bool {
```

Comment:
Lifted this straight from LXD.

Reply:
This really feels like it should be apart of the dqlite go library.

Reply:
What do you think @mathieuBordere?

Reply:
I agree that `go-dqlite` is probably better suited to determine this, tracking it [here](https://github.com/canonical/go-dqlite/issues/220)